net/strongswan/files/swanctl.init

```diff
@@ -364,7 +388,8 @@ config_child() {
 	[ -n "$if_id" ] && { swanctl_xappend4 "if_id_in = $if_id" ; swanctl_xappend4 "if_id_out = $if_id" ; }
 	[ -n "$startaction" -a "$startaction" != "none" ] && swanctl_xappend4 "start_action = $startaction"
 	[ -n "$closeaction" -a "$closeaction" != "none" ] && swanctl_xappend4 "close_action = $closeaction"
-	swanctl_xappend4 "esp_proposals = $esp_proposal"
+	[ -n "$ah_proposal" ] && swanctl_xappend4 "ah_proposals = $ah_proposal"
```

Comment:
Don't you need a check to make sure that one, or the other, is set but not both?

Reply:
From what I have read it is possible to use AH and ESP in parallel with different algorithms. I don't know why people should do this, but I think it's better to reflect the swanctl.conf rather than having users complain in the future and having to add options.

Reply:
strongSwan does not support AH+ESP bundles. The daemon detects whether ESP or AH should be negotiated for the CHILD_SA by the presence of ah_proposals (if that is defined, it's AH, otherwise it's ESP).